src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageReferenceSet.cs

```diff
@@ -13,7 +13,11 @@ namespace NuGet.Packaging
 {
     public class PackageReferenceSet
     {
-        private static readonly char[] _referenceFileInvalidCharacters = Path.GetInvalidFileNameChars();
+        internal static readonly char[] ReferenceFileInvalidCharacters = new[] {
```

Comment:
Seems like this can reference ManifestFile._invalidSourceCharacters plus these chars: ':', '*', '?', '\\', '/'.
But with the references back and forth between constants, you can put them in one place.